dotCMS/src/main/java/com/dotcms/ema/proxy/ProxyTool.java

```diff
@@ -0,0 +1,1214 @@
+                }
+            } else if (method.equalsIgnoreCase(METHOD_HEAD)) {
+                HttpHead m = new HttpHead(url + urlParamsSB.toString());
+                try(CloseableHttpResponse r = client.execute(m)){
+                    return new ProxyResponse(r.getStatusLine(), EntityUtils.toByteArray(r.getEntity()), r.getAllHeaders());
+                }
+            } else if (method.equalsIgnoreCase(METHOD_DELETE)) {
+                HttpDelete m = new HttpDelete(url + urlParamsSB.toString());
+                try(CloseableHttpResponse r = client.execute(m)){
+                    return new ProxyResponse(r.getStatusLine(), EntityUtils.toByteArray(r.getEntity()), r.getAllHeaders());
+                }
+            } else if (method.equalsIgnoreCase(METHOD_GET)) {
+                HttpGet m = new HttpGet(url + urlParamsSB.toString());
+                try(CloseableHttpResponse r = client.execute(m)){
+                    return new ProxyResponse(r.getStatusLine(), EntityUtils.toByteArray(r.getEntity()), r.getAllHeaders());
+                }
+            } else {
+                Logger.error(this, "Unimplemented Method: " + method);
+                return new ProxyResponse(ERR_CODE_UNIMPLEMENTED_METHOD, null, new Header[0]);
+            }
+
+        } catch (Exception e) {
+            Logger.warn(this, "Exception posting to url: " + url, e);
+            return new ProxyResponse(ERR_CODE_UNKNOWN_ERR, null, new Header[0]);
+        }
+
+    }
+
+    /**
+     * Sends a request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param method - The Method (POST/GET/PUT/HEAD/DELETE)
+     * @param contentType - The content type for this request
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse send(String url, Map<String, String> params, String method, String contentType) {
+        return send(url, params, method, contentType, null);
+    }
+
+    /**
+     * Sends a request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param method - The Method (POST/GET/PUT/HEAD/DELETE)
+     * @param contentType - The content type for this request
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object @ from URLDecoder.decode
+     */
+    public ProxyResponse send(String url, String params, String method, String contentType, Credentials creds) {
+        Logger.debug(this, "send(String) called with url=" + url + ", params=" + params + ", and method=" + method + ", contentType = "
+                + contentType + ", creds=" + UtilMethods.isSet(creds));
+
+        Map<String, String> queryPairs = new LinkedHashMap<String, String>();
+        if (params.length() > 0) {
+            String[] pairs = params.split("&");
+            for (String pair : pairs) {
+                int idx = pair.indexOf("=");
```

Comment:
Codacy found an issue: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)